### PR TITLE
Eliminate hanging processes left after `rpc/test_router.py` tests

### LINKED ISSUE #4165: test_router.py breaks b0.19 because it leaves zombies
https://buildbot.golem.network/buildbot/#/builders/9/builds/11624

Above build is completed but still timed out, when looking at the build-agent i see a process is stuck:
Screenshot 2019-05-09 at 10 10 12

### PR DESCRIPTION
**Pull request by shadeofblue**
closes #4165

### CODE REVIEW

File: tests/golem/rpc/test_router.py [hunk 1]
```diff
@@ -120,9 +123,27 @@ def format_errors(self):
...
+        p = psutil.Process()
+        return set([c.pid for c in p.children(recursive=True)])
+
     def setUp(self):
         super().setUp()
         self.state = _TestRouter.State()
+        self._children_on_start = self._get_process_children()
+
+    def tearDown(self):
+        super().tearDown()
```
Comment (maaktweluit) [inline tests/golem/rpc/test_router.py]: i think its better to teardown at the end

File: tests/golem/rpc/test_router.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# pylint: disable=protected-access
+    # pylint: disable=protected-access
```
Comment (maaktweluit) [inline tests/golem/rpc/test_router.py]: why are these spaces here?

**Review by maaktweluit — APPROVED** [review]
LGTM!

Nice fixes! some small comments, nothing blocking

File: tests/golem/rpc/test_router.py (hunk 2 above)
Comment (shadeofblue) [reply tests/golem/rpc/test_router.py]: ugh, that's probably some artifact...

File: tests/golem/rpc/test_router.py (hunk 1 above)
Comment (shadeofblue) [reply tests/golem/rpc/test_router.py]: hmm... it shouldn't matter - but particularly, in this case, I think it should be called first -> as there is stuff that touches the reactor et al in the original teardown

I wanted to be absolutely sure that we're doing our own cleanup as the very last thing before the test finishes

**Comment by mfranciszkiewicz** [comment]
What do you think on creating a test fixture from this cleanup code?

File: tests/golem/rpc/test_router.py (hunk 1 above)
Comment (maaktweluit) [reply tests/golem/rpc/test_router.py]: ok, sounds good to me!

**Review by mfranciszkiewicz — APPROVED** [review]

**Pull request merged**